app/assets/javascripts/discourse/app/components/topic-drafts-dropdown.gjs

```diff
@@ -0,0 +1,100 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { fn } from "@ember/helper";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import DropdownMenu from "discourse/components/dropdown-menu";
+import DiscourseURL from "discourse/lib/url";
+import { i18n } from "discourse-i18n";
+import DMenu from "float-kit/components/d-menu";
+
+export default class TopicDraftsDropdown extends Component {
+  @service currentUser;
+  @service composer;
+
+  @tracked drafts = [];
+
+  get showDraftMenu() {
+    return this.args.showDraftsMenu;
+  }
+
+  get otherDraftsCount() {
+    return this.args.otherDrafts > 0
+      ? i18n("drafts.dropdown.other_drafts", { count: this.args.otherDrafts })
+      : "";
+  }
+
+  @action
+  onRegisterApi(api) {
+    this.dMenu = api;
+  }
+
+  @action
+  async onShowMenu() {
+    const draftsStream = this.currentUser.get("userDraftsStream");
+    draftsStream.reset();
+
+    await draftsStream.findItems(this.site);
+    this.drafts = draftsStream.content.slice(0, this.args.draftLimit);
+  }
+
+  @action
+  async resumeDraft(draft) {
+    this.dMenu.close();
```

Comment:
I believe this is an `async` function: https://github.com/discourse/discourse/blob/690ffcf1d53079cfea8283bf22f30b82d961540c/app/assets/javascripts/float-kit/addon/lib/d-menu-instance.js#L57